packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -1,43 +1,62 @@
 /**
  * WordPress dependencies
  */
+import { useSelect, useDispatch } from '@wordpress/data';
 import { useState, useRef, useEffect } from '@wordpress/element';
 
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
 const { clearTimeout, setTimeout } = window;
-const noop = () => {};
 const DEBOUNCE_TIMEOUT = 200;
+export const highlightBlock = { selectedBlock: 1, parent: 2 };
```

Comment:
There's nothing _wrong_ with this choice of type, but it seems a bit unconventional. I might expect something more like:

```js
useDebouncedShowGestures( { highlightedBlock: 'selectedBlock' | 'parent' } )
```

Trying to simplify further: AFAIK the consumers of `useDebouncedShowGestures` always specify a type of `props.highlightedBlock` (i.e. there is never _no_ block to highlight). So what if the interface were:

```js
function useDebouncedShowGestures( { highlightParent = false } ) {
```

This would mean that it's assumed that some block is always highlighted, and by default it's the selected one. If you don't like this approach because it's less explicit, I understand. :) 

And same for the first part of my comment: feel free to ignore.

Reply:
Updated.